File: app/src/main/java/droiddevelopers254/droidconke/database/dao/StarredSessionDao.java
package droiddevelopers254.droidconke.database.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import droiddevelopers254.droidconke.database.entities.StarredSessionEntity;
import droiddevelopers254.droidconke.models.StarredSessionModel;
import io.reactivex.Single;

import static androidx.room.OnConflictStrategy.REPLACE;

@Dao
public interface StarredSessionDao {

    @Query("SELECT * FROM starredSessions")
    Single<List<StarredSessionModel>> getStarredSessions();

    @Insert(onConflict = REPLACE)
    void starSession(StarredSessionModel starredSessionModel);

    @Query("UPDATE  starredSessions SET starred=:starred WHERE documentId =:documentId")
    void unStarSession(boolean starred,String documentId);

    @Query("SELECT starred FROM starredSessions WHERE documentId LIKE :documentId ")
    boolean isSessionStarred(String documentId);

}

File: app/src/main/java/droiddevelopers254/droidconke/database/dao/StarredSessionDao.kt
package droiddevelopers254.droidconke.database.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy.REPLACE
import androidx.room.Query
import droiddevelopers254.droidconke.models.StarredSessionModel

@Dao
interface StarredSessionDao {

    @Query("SELECT * FROM starredSessions")
    suspend fun starredSessions(): List<StarredSessionModel>

    @Insert(onConflict = REPLACE)
    fun starSession(starredSessionModel: StarredSessionModel)

    @Query("UPDATE  starredSessions SET isStarred=:starred WHERE documentId =:documentId")
    fun unStarSession(starred: Boolean, documentId: String)

    @Query("SELECT isStarred FROM starredSessions WHERE documentId LIKE :documentId ")
    fun isSessionStarred(documentId: String): Boolean

}
